src/tests/test_relations.rs

```diff
@@ -199,27 +199,50 @@ pub fn pedersen_commitment_dleq<G: PrimeGroup, R: RngCore>(
     (instance, witness_vec)
 }
 
-/// Test that a Pedersen commitment is between 0 and 1337.
+/// Test that a Pedersen commitment is in `[0, bound)` for any `bound >= 0`.
 #[allow(non_snake_case)]
-pub fn test_range<G: PrimeGroup, R: RngCore>(
+pub fn test_range_for_input_and_bound<G: PrimeGroup, R: RngCore>(
     mut rng: &mut R,
+    input: u64,
+    bound: u64,
 ) -> (CanonicalLinearRelation<G>, Vec<G::Scalar>) {
     let G = G::generator();
     let H = G::random(&mut rng);
 
-    let bases = [1, 2, 4, 8, 16, 32, 64, 128, 256, 313, 512].map(G::Scalar::from);
-    const BITS: usize = 11;
+    let bits = {
+        let mut bits = bound.ilog2();
+        if 1 << bits < bound {
+            bits += 1;
+        }
+        usize::try_from(bits).unwrap()
+    };
+
+    // Compute the bases used to express the input as a linear combination of the bit decomposition
+    // of the input.
+    let bases = (0..bits - 1)
+        .map(|i| 1 << i)
+        .chain(Some(bound - (1 << (bits - 1))))
+        .map(G::Scalar::from)
+        .collect::<Vec<_>>();
```

Comment:
I _think_ this is correct, but just to sanity check - can we make sure that the result when the bits are a vector of all 1's equals `bound - 1`? I'd hate to have an off-by-one error somewhere in these computations.
In other words, a check that `bases.sum() == bound - 1`?